Hostap module glue code

jukkar opened the pull request:
This provides interface glue code to hostap module in `modules/hostap` directory. Crypto support is disabled.
The example native_sim based example / testing wifi driver is still WIP and is sent in a separate PR after this one is merged.

krish2718 committed:
hostap: Add Zephyr support

Add new files from hostap to Zephyr build.
Add wifi mgmt thread that communicates with supplicant.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>
Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
manifest: Update hostap information

New hostap version available.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
wifi: Have a separate wifi_drv_ops in wifi_mgmt_offload struct

There is no need to hide the wpa_supplicant interface
struct so add it directly to net_wifi_mgmt_offload struct.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
hostap: Various changes to the API to wpa_supplicant

Changed the names of the API functions between wpa_supplicant
and Zephyr upper layers. For example replaced the z_ prefix
by zephyr_ as the z_ is only meant for internal kernel
functions.

Changed the wpa_supplicant startup so that we do not poll
but wait the network interface add event which then adds
the network interfaces to the supplicant.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
hostap: Align to Zephyr coding style

Tweaked the code to be more readable and align to
Zephyr coding style.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar commented:
This update adds scanning support.

jukkar commented:
Added initial `wifi connect` support. The connection is initiated but something is still wrong as the connection status is not set properly.

MaochenWang1 commented:
Hi @jukkar , I am from NXP wifi team, when will you merge these wpa-supplicant changes into Zephyr?

jukkar commented:
> I am from NXP wifi team, when will you merge these wpa-supplicant changes into Zephyr?

We would like to get more feedback from people about the PR. But if everything looks ok (please give feedback here), I can certainly start to prepare a version that is suitable for merging. There is the related PR at https://github.com/zephyrproject-rtos/hostap/pull/2 that has the hostap changes also waiting a review.

So I will most probably remove the native_sim wifi testing driver from this PR as it requires still more work and I can send it separately after the wpa_supplicant is merged.

jukkar committed:
hostap: Avoid DHCPv4 related warnings if DHCPv4 is disabled

If the DHCPv4 is disabled, then avoid linking warning about
missing DHCPv4 functions. Also include dhcpv4.h so that the
DHCPv4 function prototypes are found by the zephyr supplicant
driver.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
hostap: Change the name of the control api struct

Zephyr requires that the name of the variable is not the
same as the struct as seen by this warning:

modules/hostap/src/supp_api.c:64:WARNING:
  Violation to rule 5.7 (Tag name should be unique)
  tag: wpa_supp_api_ctrl

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar commented:
Prepared a new version fixing CI. Also removed the native_sim wifi driver as it requires more work and can be submitted separately.

jukkar committed:
hostap: Replace the command line cmd macro

Zephyr compliance checker does not like macros that have control
structures like goto. So change the macro to align with this rule.
Also rename the macro and remove the __ prefix as it not needed.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
hostap: Temporary kconfig options to pass the compliance checker

Compliance checker does not allow compile definitions without
a Kconfig option. So add dummy values for time being. These should
be removed at some point and replaced with properly namespaced
options.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar commented:
I accidentally removed the DNM label as the corresponding hostap PR needs to be merged first, so adding it back.

krish2718 commented on modules/hostap/src/supp_main.c:
```diff
@@ -0,0 +1,546 @@
+#include <l2_packet/l2_packet.h>
+
+/* Should match with the driver name */
+#define DEFAULT_IFACE_NAME "wlan0"
```
Should we make this configurable?

jukkar replied:
This was also gone already, the correct interface is detected automatically.

krish2718 commented on modules/hostap/src/supp_main.c:
```diff
@@ -0,0 +1,546 @@
+
+DEFINE_WIFI_NM_INSTANCE(wpa_supplicant, &wpa_supp_ops);
+
+#ifdef CONFIG_MATCH_IFACE
```
This isn't used and not really applicable for Zephyr, so, can be removed.

jukkar replied:
Hi @krish2718, just asking that did you review some older version because I do not have that code in my tree, it was removed already in the latest update.

krish2718 replied:
Ah you are right, its gone now :).

krish2718 commented on modules/hostap/Kconfig:
```diff
@@ -8,7 +8,11 @@
 	select POSIX_CLOCK
+	select POSIX_SIGNAL
+	select POSIX_API
+#	depends on !POSIX_API
```
Does this work fine? I was seeing issues with POSIX implementation in Zephyr?

jukkar replied:
Should work now ok when using picolibc

krish2718 commented on modules/hostap/Kconfig:
```diff
@@ -51,6 +59,114 @@ config WIFI_NM_WPA_SUPPLICANT_DEBUG_LEVEL
+	select PSA_WANT_KEY_TYPE_RSA_KEY_PAIR
+	select PSA_WANT_ECC_SECP_R1_256
+	# Legacy crypto, still needed
+	#select MBEDTLS
```
Please remove commented Kconfigs

jukkar replied:
Good point, will do.

jukkar committed:
hostap: Remove commented Kconfig options

Some crypto Kconfig options were commented out, so remove them.
These will be re-introduced later when adding crypto support
properly.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar commented:
manifest file changed as the hostap module was updated
